### DISCUSSION TITLE
In Terrramaster NAS: Stuck in setup, behind NPM, mastercontainer give a 502 error

### DISCUSSION THREAD

**Discussion by Tab3r** [nextcloud/all-in-one#3038] [Questions] [answered]
Hi, this is my scenario:

Docker + Portainer. I created a user bridge network, for name host resolve... I can't use host network for security issues.

I use NPM as reverse proxy, with other instance of nextcloud in RaspPi without problems (without AIO).

I launch AIO with the _docker-compose_ example in the proxy reverse documentation, without issues in docker. But when i try to access to "https://MY_IP:8080", the container respond with a 502 and the Apache error:

**Reason: DNS lookup failure for: localhost**

In a shell in "nextcloud-aio-mastercontainer", i can resolve without problems "localhost" (i installed dig in container to test this issue), with nslookup and dig. Logs from container show the follow message:

**[Wed Jul 26 17:12:36.109512 2023] [proxy:error] [pid 710:tid 140469692144440] [client REDACTED:46454] AH00898: DNS lookup failure for: localhost returned by /**

I tried several hours without success... please, please, help :)

**Comment by szaimen** [comment]
Hi, can you post the complete mastercontainer logs here please?

**Reply by Tab3r** [reply]
Hi, first: Thank you for your time!

Here is the log from a fresh deployed mastercontainer, and after i tried to access to **https://10.1.30.22:8080** :

```
Trying to fix docker.sock permissions internally...
Creating docker group internally with id 1008
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
Initial startup of Nextcloud All-in-One complete!
You should be able to open the Nextcloud AIO Interface now on port 8080 of this server!
E.g. https://internal.ip.of.this.server:8080
If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443
INF ts=1690485608.8429224 msg=using provided configuration config_file=/Caddyfile config_adapter=
Error: loading initial config: loading new config: starting caddy administration endpoint: listen tcp: lookup localhost on [::1]:53: dial udp [::1]:53: connect: cannot assign requested address
[Thu Jul 27 19:20:08.857665 2023] [mpm_event:notice] [pid 105:tid 139942824647496] AH00489: Apache/2.4.57 (Unix) OpenSSL/3.1.1 configured -- resuming normal operations
[Thu Jul 27 19:20:08.858124 2023] [core:notice] [pid 105:tid 139942824647496] AH00094: Command line: 'httpd -D FOREGROUND'
[27-Jul-2023 19:20:08] NOTICE: fpm is running, pid 109
[27-Jul-2023 19:20:08] NOTICE: ready to handle connections
INF ts=1690485609.8905277 msg=using provided configuration config_file=/Caddyfile config_adapter=
Error: loading initial config: loading new config: starting caddy administration endpoint: listen tcp: lookup localhost on [::1]:53: dial udp [::1]:53: connect: cannot assign requested address
INF ts=1690485611.929471 msg=using provided configuration config_file=/Caddyfile config_adapter=
Error: loading initial config: loading new config: starting caddy administration endpoint: listen tcp: lookup localhost on [::1]:53: dial udp [::1]:53: connect: cannot assign requested address
INF ts=1690485614.9687974 msg=using provided configuration config_file=/Caddyfile config_adapter=
Error: loading initial config: loading new config: starting caddy administration endpoint: listen tcp: lookup localhost on [::1]:53: dial udp [::1]:53: connect: cannot assign requested address
[Thu Jul 27 19:20:40.531257 2023] [proxy:error] [pid 122:tid 139942813145912] [client 10.1.30.15:54272] AH00898: DNS lookup failure for: localhost returned by /
[Thu Jul 27 19:20:45.710513 2023] [proxy:error] [pid 122:tid 139942813002552] [client 10.1.30.15:54272] AH00898: DNS lookup failure for: localhost returned by /favicon.ico, referer: https://10.1.30.22:8080/
[Thu Jul 27 19:20:54.195781 2023] [proxy:error] [pid 122:tid 139942812859192] [client 10.1.30.15:54272] AH00898: DNS lookup failure for: localhost returned by /
[Thu Jul 27 19:20:59.360740 2023] [proxy:error] [pid 122:tid 139942812715832] [client 10.1.30.15:54272] AH00898: DNS lookup failure for: localhost returned by /favicon.ico, referer: https://10.1.30.22:8080/
``` 
Again, thank you very much!

**Reply by szaimen** [reply]
Hi, I googled your issue and found this: https://stackoverflow.com/a/62200923

**Reply by Tab3r** [reply]
Thank you!... i will try tomorrow setting the dns to the mastercontainer... I'll let you know

**Reply by Tab3r** [reply]
No luck... still same 502 error. But nslookup work without problem, resolving the host.

I tried several changes:
- I set in `/etc/resolv.conf`the nameserver to 1.1.1.1, and 8.8.8.8.
- I set the nameserver to my local dns server (pihole), same error.
- I set a hostname to the container, and change all localhost strings in `/etc/apache2/sites-available/mastercontainer.conf` for the hostname. Same error, 502, but now says `AH00898: DNS lookup failure for: mastercontainer returned by /`.
- I tried to remove ipv6 support... nothing changes.

Inside the logs, the line about Caddy:

```
Error: loading initial config: loading new config: starting caddy administration endpoint: listen tcp: lookup localhost on [::1]:53: dial udp [::1]:53: connect: cannot assign requested address
```
It could be an error or mismatch with ipv4 vs ipv6?...

I'm still working, on the docker parameters, and i stay "tunning" the linux inside the mastercontainer to find a solution (i will share here if i find one).

Thank you, again!

**Comment by Tab3r** [comment, marked as answer]
hello, at last, I have found the problem.

In short: Terramaster NAS

I have Docker running on a Terramaster NAS, with its official package and the problem comes from some error, with permissions. It is a bug reported and not fixed.

The problem is that Docker's internal DNS server doesn't work correctly, if you use a user defined network.

Here are the links in case anyone is interested:

- https://forum.terra-master.com/en/viewtopic.php?f=43&t=3817&start=20
- https://forum.terra-master.com/en/viewtopic.php?f=43&t=4881
- https://forum.terra-master.com/en/viewtopic.php?f=43&t=4947

I'm going to change the text, to make the error clearer.

Sorry for the inconvenience!
